### ISSUE TITLE
environment problem

### ISSUE THREAD

**Issue by Korace0v0** [astra-vision/SceneRF#11]
Hi, I met a bug in DecoderSphere.
        feats = F.grid_sample(
            x,
            map_sphere,
            align_corners=False,
            mode='bilinear'
        )
grid_sampler(): expected grid and input to have same batch size, but got input with sizes [4, 2560, 14, 41] and grid with sizes [1, 1, 658, 2].

Here is my enviroment information. Is there anything wrong? I truly appreciate your help.
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
absl-py                   1.4.0              pyhd8ed1ab_0    conda-forge
addict                    2.4.0                    pypi_0    pypi
aiohttp                   3.7.0            py37h8f50634_0    conda-forge
anyio                     3.6.2                    pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
async-timeout             3.0.1                   py_1000    conda-forge
attrs                     22.2.0             pyh71513ae_0    conda-forge
backcall                  0.2.0                    pypi_0    pypi
beautifulsoup4            4.11.2                   pypi_0    pypi
blas                      1.0                         mkl    defaults
bleach                    6.0.0                    pypi_0    pypi
blinker                   1.5                pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py37h540881e_1004    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2023.01.10           h06a4308_0    defaults
cachetools                5.3.0              pyhd8ed1ab_0    conda-forge
certifi                   2022.12.7        py37h06a4308_0    defaults
cffi                      1.15.1                   pypi_0    pypi
chardet                   3.0.4           py37he5f6b98_1008    conda-forge
charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
click                     8.1.3            py37h89c1867_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
cryptography              37.0.2           py37h38fbfac_0    conda-forge
cudatoolkit               11.0.221             h6bb024c_0    defaults
cycler                    0.11.0                   pypi_0    pypi
debugpy                   1.6.6                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
fastjsonschema            2.16.3                   pypi_0    pypi
flit-core                 3.6.0              pyhd3eb1b0_0    defaults
fonttools                 4.38.0                   pypi_0    pypi
freetype                  2.12.1               h4a9f257_0    defaults
fsspec                    2023.1.0           pyhd8ed1ab_0    conda-forge
future                    0.18.2           py37h89c1867_5    conda-forge
giflib                    5.2.1                h5eee18b_3    defaults
google-auth               2.16.1             pyh1a96a4e_0    conda-forge
google-auth-oauthlib      0.4.6              pyhd8ed1ab_0    conda-forge
grpcio                    1.38.1           py37hb27c1af_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
imageio                   2.26.0                   pypi_0    pypi
importlib-metadata        6.0.0                    pypi_0    pypi
importlib-resources       5.12.0                   pypi_0    pypi
intel-openmp              2021.4.0          h06a4308_3561    defaults
ipykernel                 6.16.2                   pypi_0    pypi
ipython                   7.34.0                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                8.0.4                    pypi_0    pypi
jedi                      0.18.2                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
joblib                    1.2.0                    pypi_0    pypi
jpeg                      9b                   h024ee3a_2    defaults
jsonschema                4.17.3                   pypi_0    pypi
jupyter-client            7.4.9                    pypi_0    pypi
jupyter-core              4.12.0                   pypi_0    pypi
jupyter-server            1.23.6                   pypi_0    pypi
jupyterlab-pygments       0.2.2                    pypi_0    pypi
jupyterlab-widgets        3.0.5                    pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
lcms2                     2.12                 h3be6417_0    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libffi                    3.4.2                h6a678d5_6    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libpng                    1.6.37               hbc83047_0    defaults
libprotobuf               3.18.0               h780b84a_1    conda-forge
libstdcxx-ng              11.2.0               h1234567_1    defaults
libtiff                   4.1.0                h2733197_1    defaults
libuv                     1.44.2               h5eee18b_0    defaults
libwebp                   1.2.0                h89dd481_0    defaults
llvmlite                  0.36.0                   pypi_0    pypi
lpips                     0.1.4                    pypi_0    pypi
lz4-c                     1.9.4                h6a678d5_0    defaults
markdown                  3.4.1              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.2                    pypi_0    pypi
matplotlib                3.5.3                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mistune                   2.0.5                    pypi_0    pypi
mkl                       2021.4.0           h06a4308_640    defaults
mkl-service               2.4.0            py37h7f8727e_0    defaults
mkl_fft                   1.3.1            py37hd3c417c_0    defaults
mkl_random                1.2.2            py37h51133e4_0    defaults
multidict                 6.0.2            py37h540881e_1    conda-forge
nbclassic                 0.5.2                    pypi_0    pypi
nbclient                  0.7.2                    pypi_0    pypi
nbconvert                 7.2.9                    pypi_0    pypi
nbformat                  5.7.3                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0    defaults
nest-asyncio              1.5.6                    pypi_0    pypi
networkx                  2.6.3                    pypi_0    pypi
ninja                     1.10.2               h06a4308_5    defaults
ninja-base                1.10.2               hd09550d_5    defaults
notebook                  6.5.2                    pypi_0    pypi
notebook-shim             0.2.2                    pypi_0    pypi
numba                     0.53.0                   pypi_0    pypi
numpy                     1.20.3                   pypi_0    pypi
oauthlib                  3.2.2              pyhd8ed1ab_0    conda-forge
open3d                    0.12.0                   pypi_0    pypi
open3d-python             0.7.0.0                  pypi_0    pypi
opencv-python             4.5.1.48                 pypi_0    pypi
openssl                   1.1.1t               h7f8727e_0    defaults
packaging                 23.0               pyhd8ed1ab_0    conda-forge
pandas                    1.3.5                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.3.0            py37hace64e9_1    defaults
pip                       22.3.1           py37h06a4308_0    defaults
pkgutil-resolve-name      1.3.10                   pypi_0    pypi
plyfile                   0.7.4                    pypi_0    pypi
prometheus-client         0.16.0                   pypi_0    pypi
prompt-toolkit            3.0.38                   pypi_0    pypi
protobuf                  3.18.0           py37hcd2ae1e_0    conda-forge
psutil                    5.9.4                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pyasn1                    0.4.8                      py_0    conda-forge
pyasn1-modules            0.2.7                      py_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydeprecate               0.3.1              pyhd8ed1ab_0    conda-forge
pygments                  2.14.0                   pypi_0    pypi
pyjwt                     2.6.0              pyhd8ed1ab_0    conda-forge
pyopenssl                 22.0.0             pyhd8ed1ab_1    conda-forge
pyparsing                 3.0.9                    pypi_0    pypi
pyrsistent                0.19.3                   pypi_0    pypi
pysocks                   1.7.1            py37h89c1867_5    conda-forge
python                    3.7.16               h7a1cb2a_0    defaults
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.7                     2_cp37m    conda-forge
pytorch                   1.7.1           py3.7_cuda11.0.221_cudnn8.0.5_0    pytorch
pytorch-lightning         1.4.9              pyhd8ed1ab_0    conda-forge
pytz                      2022.7.1                 pypi_0    pypi
pyu2f                     0.1.5              pyhd8ed1ab_0    conda-forge
pywavelets                1.3.0                    pypi_0    pypi
pyyaml                    6.0              py37h540881e_4    conda-forge
pyzmq                     25.0.0                   pypi_0    pypi
readline                  8.2                  h5eee18b_0    defaults
requests                  2.28.2             pyhd8ed1ab_0    conda-forge
requests-oauthlib         1.3.1              pyhd8ed1ab_0    conda-forge
rsa                       4.9                pyhd8ed1ab_0    conda-forge
scenerf                   0.0.0                     dev_0    <develop>
scikit-image              0.18.1                   pypi_0    pypi
scikit-learn              0.24.0                   pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
send2trash                1.8.0                    pypi_0    pypi
setuptools                65.6.3           py37h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sklearn                   0.0.post1                pypi_0    pypi
sniffio                   1.3.0                    pypi_0    pypi
soupsieve                 2.4                      pypi_0    pypi
sqlite                    3.40.1               h5082296_0    defaults
tbb                       2020.2               hff7bd54_0    defaults
tensorboard               2.11.2             pyhd8ed1ab_0    conda-forge
tensorboard-data-server   0.6.0            py37h38fbfac_2    conda-forge
tensorboard-plugin-wit    1.8.1              pyhd8ed1ab_0    conda-forge
terminado                 0.17.1                   pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2021.11.2                pypi_0    pypi
tinycss2                  1.2.1                    pypi_0    pypi
tk                        8.6.12               h1ccaba5_0    defaults
torchaudio                0.7.2                      py37    pytorch
torchmetrics              0.6.0                    pypi_0    pypi
torchvision               0.8.2                py37_cu110    pytorch
tornado                   6.2                      pypi_0    pypi
tqdm                      4.49.0                   pypi_0    pypi
traitlets                 5.9.0                    pypi_0    pypi
typing_extensions         4.4.0            py37h06a4308_0    defaults
urllib3                   1.26.14            pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.6                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.5.1                    pypi_0    pypi
werkzeug                  2.2.3              pyhd8ed1ab_0    conda-forge
wheel                     0.38.4           py37h06a4308_0    defaults
widgetsnbextension        4.0.5                    pypi_0    pypi
xz                        5.2.10               h5eee18b_1    defaults
yaml                      0.2.5                h7f98852_2    conda-forge
yarl                      1.6.0            py37h8f50634_0    conda-forge
zipp                      3.15.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h5eee18b_0    defaults
zstd                      1.4.9                haebb681_0    defaults

**Comment by anhquancao** [comment]
Hello @zhangfang97,
Thank you for your interest!
At present, the code only allows for training one item per GPU using DDP. Your input has a batch dimension of 4, while the map_sphere batch dimension 0 is 1.

**Comment by Korace0v0** [comment]
Thank you very much! I got it. This is really a great work. 😊

**Comment by anhquancao** [comment]
Thank you very much for your kind words! I really appreciate it!
